non-normative/acpi.adoc

```diff
@@ -1,7 +1,8 @@
 ACPI information is structured as tables with the address of the root of these
 tables known as Root System Description Table (RSDP) passed to the OS
-via a EFI_ACPI_20_TABLE_GUID configuration table in the UEFI firmware.
-The Operating System uses this address to locate all other ACPI tables.
+via a EFI_ACPI_20_TABLE_GUID configuration table in the UEFI firmware. In
+addition, RSDP can be obtained in other ways, such as DT. The Operating System
+uses this address to locate all other ACPI tables.
```

Comment:
This is very vague. Where are these other ways defined?

Reply:
We already have a whole set of schemes based on device tree, and have completed the preliminary implementation and verification. This part of the solution is also being prepared upstream.

Reply:
Hate to be a pedant, but the point of a spec is exactly "to avoid a whole set of schemes", and you should have the mechanism well identified and described in the spec as part of your pull request, not as a promise. As it stands if someone were to implement your pull request today, they couldn't do it.

Reply:
We are now promoting and improving this recipe into a mechanism. As for DT-based passing mechanisms, we are also pushing DT spec to support ACPI/SMBIOS pointer passing.

Reply:
> We are now promoting and improving this recipe into a mechanism. As for DT-based passing mechanisms, we are also pushing DT spec to support ACPI/SMBIOS pointer passing.

It is probably worth noting that I NAKed the DT side of this, as it violated "UEFI firmware is mandatory to support ACPI" in https://github.com/riscv-non-isa/riscv-acpi/blob/master/riscv-acpi-guidance.adoc. Not gonna be changing my mind on that unless the specs get changed first.

Reply:
I've already said it: this spec is obsolete and no one maintains it, and this guidance was only written by sunil, which was also discussed with sunil @vlsunil 

How about deleting this guidance? So as not to cause misunderstanding.

Reply:
That's right. The guidance document was created long back when platform spec was being drafted. This was never intended to be a spec and neither ratified/frozen. Also, now things have changed to BRS spec which has more official version of the similar requirements. The guidance document  has old things like only AIA will be supported with ACPI etc which is no longer the case. So, there is no value in keeping this any more.  I will be deleting this guidance document soon in favor of BRS spec.

Reply:
FWIW, that doesn't change my position. I don't care where the rules are defined, but unless they clearly say that UEFI is not required to use ACPI the NAK stands.

Reply:
ACPI ans UEFI were two seperated spec before, both are interfaces between FW and OS, and ACPI was moved to UEFI org since ARM platform need to support both UEFI and ACPI; in the ACPI soec 5.2.5, it defined RSDP and also list two solutions how to find the RSDP on IA-PC and UEFI edabled systems, they are for reference ony. 
With current solution, it's on non-UEFI enabled system, and there is NOT EBDA on non-x86 platform, so I think this is great solution to find the RSDP from DTS.

Reply:
Sorry, I don't agree that the mechanisms mentioned in ACPI spec to find the RSDP is only for reference. Because that will imply that the platforms are free to pass RSDP in any arbitrary manner and OSPMs can find them which is not true. The FW and OSs need a known contract and that's why those two mechanisms are documented in the ACPI spec. Same applies to SMBIOS spec also.

I fail to understand why are you reluctant to propose change to ACPI/SMBIOS specs. They are the original place where UEFI is mandated. The proposal has nothing really which applies to only RISC-V. So,if you want to remove UEFI requirement, ACPI/SMBIOS specs should be updated with new mechanism to discover ACPI/SMBIOS.

Reply:
In Chrome book, how could firmware pass the ACPI and SMBIOS points to Chrome OS? and could you share with us which spec has cover it?

Reply:
Sorry I don't know about chromeOS. I am keen to learn that it doesn't use either of the ways defined by the ACPI spec (EBDA or UEFI). Could you find the answer and provide more information of the mechanism?

Reply:
From my working on Chrome OS only x86 systems utilized ACPI. I'm assuming that's still true. On x86 the kernel goes and searches in low memory for the expected entries.

https://review.coreboot.org/plugins/gitiles/coreboot/+/refs/heads/main/src/arch/x86/tables.c

Reply:
Ron, on the "the forthcoming UPL standard will define a new one that is DT-based": is that about finding the ACPI root table? And would the kernel be changed to leverage that path as well?

Reply:
The problem with using DT to discover ACPI is the fact that DT and ACPI are both hardware descriptions, and are usually mutually exclusive. Using both at the same time forces the OS to reason about whether or not they are mutually consistent, and about which description should take precedence when they are not.

Reply:
Can the OS scan UEFI first? If it is not obtained, it can be obtained in the same way as DT.

Reply:
Currently in our FFI solution, the RSDP obtained through DT is filled into the acpi_arch_get_root_pointer() interface. The current logic is to obtain it through acpi_arch_get_root_pointer() first. If it cannot be obtained, go to the EFI entrance.

See drivers/acpi/osl.c

pa = acpi_arch_get_root_pointer();
if (pa)
return pa;

if (efi_enabled(EFI_CONFIG_TABLES)) {
if (efi.acpi20 != EFI_INVALID_TABLE_ADDR)
return efi.acpi20;
if (efi.acpi != EFI_INVALID_TABLE_ADDR)
return efi.acpi;
....

Reply:
The relationship between ACPI and DTS is compatible in the current kernel, so there is no problem.

Regarding DT transmitting ACPI,  only need to fill in RSDP. ACPI is an independent framework, and EFI and DT are just two different entry interfaces.

Reply:
This was also discussed that there needs to be alternatives documented in specs for finding ACPI tables.  I indicated that the PRS TG in RVI is the path to achieve that for updating ACPI specs.

Likewise, I also indicated that the kernel maintainers need to be on board as that was where a lot of the disagreement exists. These two things are interrelated which is why the dependencies need to be called out and completed.

Reply:
Yes, we can merge this PR into BRS spec first.
Regarding ACPI, we are also pushing forward the modification of RSDP search method. At the same time, we are also promoting and discussing linux kernel PR with the community.